### PR TITLE
ENH: added bool as for #52

### PR DESCRIPTION
**Pull request by BenKaehler**
I've added support for Bool primitive in the standard click way. There are other ways to do it, so please let me know if you would like something different.

### CODE REVIEW

**Comment by jairideout** [comment]
👍 , this looks good to me but I am not very savvy with the framework's primitive type system. @ebolyen, can you do a final review/merge?

File: q2cli/handlers.py [hunk 1]
```diff
@@ -267,9 +268,16 @@ def get_type(self):
...
+            no_name = self.prefix + 'no_' + self.name
+            cli_no_name = q2cli.util.to_cli_name(no_name)
+            yield click.Option(['--' + self.cli_name + '/--' + cli_no_name])
+        else:
+            yield click.Option(['--' + self.cli_name], type=type)
 
     def get_value(self, arguments, fallback=None):
         value = self._locate_value(arguments, fallback)
+        if self.get_type() is bool:
+            value = self.semtype.encode(value)
```
Comment (ebolyen) [inline q2cli/handlers.py]: I would probably just do an early `return value` here instead of encoding to decode.

File: q2cli/handlers.py [hunk 2]
```diff
@@ -267,9 +268,16 @@ def get_type(self):
         return mapping[self.ast['name']]
 
     def get_click_options(self):
-        yield click.Option(['--' + self.cli_name],
-                           type=self.get_type())  # Use the ugly lookup above
+        type = self.get_type()  # Use the ugly lookup above
+        if type is bool:
+            no_name = self.prefix + 'no_' + self.name
+            cli_no_name = q2cli.util.to_cli_name(no_name)
+            yield click.Option(['--' + self.cli_name + '/--' + cli_no_name])
```
Comment (ebolyen) [inline q2cli/handlers.py]: Nice!

(granted this handler is a hack, but good extension of it!)

**Comment by ebolyen** [comment]
Thanks @BenKaehler! One tiny comment, but should be good to merge afterwards.

**Comment by BenKaehler** [comment]
Thanks @ebolyen. Done!

**Comment by jairideout** [comment]
👍 , thanks @BenKaehler!

**Pull request merged**